src/main/java/algorithms/orthogonalRangeSearching/twoDim/OrthogonalRangeSearching.java

```diff
@@ -0,0 +1,314 @@
+package algorithms.orthogonalRangeSearching.twoDim;
+
+import algorithms.orthogonalRangeSearching.RangeTreeNode;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class OrthogonalRangeSearching {
+
+    /**
+     * Builds the X-tree, a data structure for the given range of 2D points.
+     *
+     * @param inputs List of 2D points in the format Integer[], where each array
+     *               represents a point with two coordinates (x, y).
+     * @param start  The starting index of the current range.
+     * @param end    The ending index of the current range.
+     * @return The root node of the X-tree for the specified range.
+     */
+    public static RangeTreeNode<Integer[]> buildXTree(List<Integer[]> inputs, int start, int end) {
```

Comment:
this method is a little long and readers might easily get lost. Suggestion is to add some brief comments to explain the major steps, like having to build the y-tree etc

Reply:
also, sanity check for myself too: the y-tree must be built at every intermediate node right?

Reply:
yeap